src/main/java/com/example/SOPSbackend/service/DoctorService.java

```diff
@@ -1,13 +1,50 @@
 package com.example.SOPSbackend.service;
 
+import com.example.SOPSbackend.exception.InternalValidationException;
+import com.example.SOPSbackend.model.Doctor;
+import com.example.SOPSbackend.model.VaccinationSlot;
 import com.example.SOPSbackend.repository.DoctorRepository;
+import com.example.SOPSbackend.repository.VaccinationSlotRepository;
 import org.springframework.stereotype.Service;
 
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+
 @Service
 public class DoctorService {
-    private final DoctorRepository repository;
+    private static final int NEW_SLOT_MIN_TIME_DIFF = 15;
+
+    private final DoctorRepository doctorRepository;
+    private final VaccinationSlotRepository vaccinationSlotRepository;
+
+    public DoctorService(DoctorRepository doctorRepository, VaccinationSlotRepository vaccinationSlotRepository) {
+        this.doctorRepository = doctorRepository;
+        this.vaccinationSlotRepository = vaccinationSlotRepository;
+    }
+
+    public void addVaccinationSlot(Doctor doctor, Instant date) {
+        LocalDateTime transformedDate = transformVaccinationSlotDate(date);
+
+        if(!isVaccinationSlotDateValid(transformedDate))
+            throw new InternalValidationException(new HashMap<String, String>() {{
+                put("date", "Invalid date value");
+            }});
+
+        VaccinationSlot newSlot = new VaccinationSlot();
```

Comment:
ja bym tego nie rozwlekał na 4 linijki tylko napisał konstruktor (który sam w sobie zajmie 4 linijki, czyli straciliśmy 1 linijkę, ale za to lepiej będzie się czytało). Ale chętnie poznam argumenty za zastosowanym podejściem @adamczykt 

IMO